### ISSUE TITLE
Rock:Grid not using ViewState for MergeTemplatePageRoute

### ISSUE THREAD

**Issue by treyhendon** [SparkDevNetwork/Rock#1614]
https://github.com/SparkDevNetwork/Rock/blob/release-1.5.0/Rock/Web/UI/Controls/Grid/Grid.cs#L432

The Rock:Grid's MergeTemplatePageRoute value should come from the ViewState, however, in 1.5.0 that is not happening.  No matter what the grid is using the null condition's value of "~/MergeTemplate/{0}"

**Comment by azturner** [comment]
@treyhendon, no where in core do we set that property to anything different. Do you have a custom block where you are setting that property to something else (so that something actually gets saved to viewstate)?

**Comment by treyhendon** [comment]
@azturner  Yes.  Tried it as an earlier set in the Batch List block (OnInit) and have now created a standalone "Set ViewState Item" block which allows me to set that ViewState value in a block before the Batch List block.  Still not working.  I'm setting the ViewState with the following code and have used the Debugger to verify the ViewState value is set, valid, and persists.

`ViewState["MergeTemplatePageRoute"] = "~/FinanceMergeTemplate/{0}";`

**Comment by nairdo** [comment]
Can you also verify you have ViewState enabled on that page (Advanced Settings)?
![image](https://cloud.githubusercontent.com/assets/383807/16689220/c901b4f0-44d6-11e6-9a04-49cabe470ddb.png)

**Comment by azturner** [comment]
@treyhendon Viewstate is specific to each control, You can't set the grid's viewstate outside of the grid. You need to set the grid's MergeTemplatePageRoute property for the grid to be able to save/read it's value from viewstate. And, you need to do it after OnInit.  Try doing it in OnLoad and let me know if you still have an issue.

**Comment by treyhendon** [comment]
Hmm. Ok, I was trying to set that property last night, but as soon as I set the target, the icon would disappear - which is what lead me down this rabbit hole. :)  I'll verify that I'm setting that property OnLoad and let you know.

Great question @nairdo but ViewState is enabled for the page.  From what @azturner is saying, I don't know that I'd ever be able to access the Grid's ViewState, would I?
